packages/angular/build/src/tools/esbuild/javascript-transformer-worker.ts

```diff
@@ -125,17 +128,10 @@ async function requiresLinking(path: string, source: string): Promise<boolean> {
     return false;
   }
 
-  if (!needsLinking) {
-    // Load ESM `@angular/compiler-cli/linker` using the TypeScript dynamic import workaround.
-    // Once TypeScript provides support for keeping the dynamic import this workaround can be
-    // changed to a direct dynamic import.
-    const linkerModule = await loadEsmModule<typeof import('@angular/compiler-cli/linker')>(
-      '@angular/compiler-cli/linker',
-    );
-    needsLinking = linkerModule.needsLinking;
-  }
-
-  return needsLinking(path, source);
+  // Check if the source code includes one of the declaration functions.
```

Comment:
NOT: Can we also add a faster check to check that the file extension is `.mjs` ? All Angular libraries are `.mjs`. This could also simplify the regexp for core and compiler.

Reply:
Third party angular libraries may not be though. I don't think APF mandates the file extension.

Reply:
Libraries that are compiled with ng-packagr is though.

Reply:
Most probably use ng-packagr but i would be hesitant to assume all do.  i think it may be good to do as a separate change in case it causes unintentional breakage.

Reply:
Also, maybe APF should mandate something like `.partial.mjs` for a more encompassing check?

Reply:
IMO APF should always be partial compilation output, but since that is not necessarily the case for 1st party libraries, I could see adding a field to the `package.json`?

Reply:
Walking up the file system to find a relevant `package.json` for each file could get expensive.
At a minimum something like `.partial.mjs` could provide a definitive signal that it needs to be processed and no additional check would be needed.  Other files could fallback to the existing checks.

Reply:
Ah, I thought it would be easy to determine the package root, and have some caching.

In either case, I think it would be easy enough to introduce the `.partial` suffix